pkg/api/agentPoolOnlyApi/v20170831/types.go

```diff
@@ -40,6 +40,19 @@ type Properties struct {
 	AccessProfiles          map[string]AccessProfile `json:"accessProfiles,omitempty"`
 }
 
+// ManagedClusterAccessProfile represents the access profile definition for managed cluster
+// The Id captures the Role Name e.g. clusterAdmin, clusterUser
+type ManagedClusterAccessProfile struct {
+	ID       string                `json:"id,omitempty"`
+	Location string                `json:"location,omitempty" validate:"required"`
+	Name     string                `json:"name,omitempty"`
+	Plan     *ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags     map[string]string     `json:"tags,omitempty"`
```

Comment:
I would just remove `Plan` and `Tags` from the API model. It's not needed and we don't want to allow setting it.